1_fetch/src/get_nwis_data.R

```diff
@@ -1,24 +1,25 @@
 
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500", "01477050"), parameterCd = '00010'){
+download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500", "01477050"), parameterCd = '00010', dl_dir = '1_fetch/out'){
   
   # create the file names that are needed for download_nwis_site_data
   # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
   # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
+  download_files <- file.path(dl_dir, paste0('nwis_', site_nums, '_data.csv'))
   
   data_out <- data.frame(agency_cd = c(), site_no = c(), dateTime = c(), 
                          X_00010_00000 = c(), X_00010_00000_cd = c(), tz_cd = c())
   # loop through files to download 
   for (download_file in download_files){
     download_nwis_site_data(download_file, parameterCd = parameterCd)
-    # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
-    data_out <- rbind(data_out, these_data)
+    
   }
-  return(data_out)
+  return(download_files)
```

Comment:
We return the vector of the downloaded files, but their names won't change if future data within them changes, so this may not trigger future rebuilds as desired.

More on this to come in the future section too.